utils/ami-cleanup/ami_cleanup.py

```diff
@@ -0,0 +1,216 @@
+    session = boto3.Session(profile_name=aws_profile)
+    asg_client = session.client("autoscaling", region_name=aws_region)
+    return asg_client
+
+
+def get_deletion_list(boto_client, ami_prefix, keep_min):
+    logger.debug(f"AMI prefix: {ami_prefix}, minimum number to keep: {keep_min}")
+    try:
+        full_list = boto_client.describe_images(
+            Filters=[{"Name": "name", "Values": [ami_prefix]}]
+        )["Images"]
+    except Exception as e:
+        logger.error(f"Failed to get list of AMIs for name prefix {ami_prefix} : {e}")
+        raise
+    sorted_list = sorted(full_list, key=lambda k: k["CreationDate"])
+    for ami in sorted_list:
+        logger.debug(f"Date: {ami['CreationDate']}, AMI Name: {ami['Name']}")
+    if len(sorted_list) > keep_min:
+        del_offset = len(sorted_list) - keep_min
+        del sorted_list[del_offset:]
+    logger.debug(f"Trimmed list:")
+    for ami in sorted_list:
+        logger.debug(f"Date: {ami['CreationDate']}, AMI Name: {ami['Name']}")
+    return sorted_list
+
+
+def check_ami_is_used(ec2_client, asg_client, ami_id):
+    # Check against running instances
+    try:
+        instance_list = ec2_client.describe_instances(
+            Filters=[{"Name": "image-id", "Values": [ami_id]}]
+        )["Reservations"]
+        if len(instance_list) > 0:
+            logger.debug(f"AMI ID: {ami_id} is used by instance(s)")
+            return True
+    except Exception as e:
+        logger.error(f"Failed to get list of Instances using AMI ID {ami_id} : {e}")
+        raise
+
+    # Check against launch configurations
+    try:
+        lc_list = asg_client.describe_launch_configurations()[
+            "LaunchConfigurations"
+        ]
+        for lc in lc_list:
+            if lc["ImageId"] == ami_id:
+                logger.debug(
+                    f"AMI {ami_id} is used by launch configuration {lc['LaunchConfigurationName']}"
+                )
+                return True
+    except Exception as e:
+        logger.error(f"Failed to get list of Launch Configurations using AMI ID {ami_id} : {e}")
+        raise
+
+    # Check against launch templates
+    try:
+        lt_list = ec2_client.describe_launch_templates()["LaunchTemplates"]
+        for lt in lt_list:
+            lt_desc = ec2_client.describe_launch_template_versions(
+                LaunchTemplateId=lt["LaunchTemplateId"],
+                Versions=["$Latest"],
+                Filters=[{"Name": "image-id", "Values": [ami_id]}],
+            )["LaunchTemplateVersions"]
+            if len(lt_desc) > 0:
+                logger.debug(
+                    f"AMI {ami_id} is used by Launch Template {lt['LaunchTemplateName']}"
+                )
+                return True
+    except Exception as e:
+        logger.error(f"Failed to get list of Launch Templates using AMI ID {ami_id} : {e}")
+        raise
+
+    return False
+
+
+def common(lst1, lst2):
+    return list(set(lst1) & set(lst2))
```

Comment:
do you not need to run `set()` on the joined lists? Can an item not exist in lst1 and lst2

Also, what do you have against vowels?